test_all

```diff
@@ -54,29 +52,56 @@ TEST_ENVS = {
     "raid4":         {"LEVEL":      "4"},
 }
 
-if "DEVS" in os.environ and os.environ["DEVS"].strip() != "":
-    del TEST_ENVS["loop"]
+def run_all_envs(tests, disk_type, env):
+    for test in tests:
+        print(f"Running {test} with {disk_type}:")
+        if args.quick:
+            run_all_quick(TEST_ENVS[test] | env)
+        else:
+            run_all_tests(TEST_ENVS[test] | env)
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument("tests", nargs="*", help="tests to run",
                         choices=list(TEST_ENVS.keys()) + [[]])
     parser.add_argument("-q", "--quick", action="store_true",
                         help="quick version of the test")
+    parser.add_argument("-d", "--disk_type", choices=("ram", "loop", "both",
+                                                      "ram_and_loop", "devs"),
+                        help="which disk types to run on")
+
     args = parser.parse_args()
 
     os.environ["ZERO_FIRST"] = "y"
     os.environ["QUIET"] = "y"
 
+    if args.disk_type is None:
+        if "DEVS" in os.enivron:
+            args.disk_type = "devs"
+        else:
+            args.disk_type = "ram"
+
+    if "DEVS" in os.environ and args.disk_type != "devs":
+        sys.exit("If DEVS is set in the environment, --disk-type must be 'devs'")
```

Comment:
Why? it's convenient to set DEVS once, and then just select the DISK_TYPE as needed when running things manually

Reply:
Yes, good point. I think I was worried that DEVS in the enviroment might change the test, but it should be fine....

Though I should check DISK_TYPE to ensure it matches....